Add Kafka heartbeat interval and session timeout to component reference (#4074)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md

| `saslMechanism` | N | Input/Output | The SASL authentication mechanism you'd like to use. Only required if `authtype` is set to `"password"`. If not provided, defaults to `PLAINTEXT`, which could cause a break for some services, like Amazon Managed Service for Kafka. | `"SHA-512", "SHA-256", "PLAINTEXT"` |
| `initialOffset`   | N | Input | The initial offset to use if no offset was previously committed. Should be "newest" or "oldest". Defaults to "newest". | `"oldest"` |
| `maxMessageBytes` | N | Input/Output | The maximum size in bytes allowed for a single Kafka message. Defaults to 1024. | `"2048"` |

> | `heartbeatInterval` | N | The interval between heartbeats to the consumer coordinator. Defaults to 3s. | `5s` |

Reviewer: Any guidance that you would consider adding here on when to change this setting?

Author: Added some guidance per confluent's documentation (https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#heartbeat-interval-ms), do you think that this comment is now also relevant to `sessionTimeout`?

> | `sessionTimeout`    | N | Input | The maximum time between heartbeats before the consumer is considered inactive and will timeout. Defaults to 10s. | `30s` |

Reviewer: Do you really mean "maximum time between heartbeats" here? Shouldn't this say "Maximum session time before all heartbeats with the heartbeatInterval stop" or similar? I would take some of the working from the Kafka documentation https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#session-timeout-ms

Author: After considering this for a bit, I do agree that it's not exactly clear, so I changed to the following (taking part of Kafka's documentation):
"The timeout used to detect client failures when using Kafka’s group management facility. If the broker fails to receive any heartbeats from the consumer before the expiration of this session timeout, then the consumer will be removed and initiate a rebalance."

I could also take most of the explanation on the `sessionTimeout` from confluent's documentation entirely instead of trying to change it.

| `clientConnectionTopicMetadataRefreshInterval` | N | Input/Output | The interval for the client connection's topic metadata to be refreshed with the broker as a Go duration. Defaults to `9m`. | `"4m"` |
| `clientConnectionKeepAliveInterval` | N | Input/Output | The maximum time for the client connection to be kept alive with the broker, as a Go duration, before closing the connection. A zero value (default) means keeping alive indefinitely. | `"4m"` |
| `consumerFetchDefault` | N | Input/Output | The default number of message bytes to fetch from the broker in each request. Default is `"1048576"` bytes. | `"2097152"` |

> | `heartbeatInterval` | N | Input | The interval between heartbeats to the consumer coordinator. The value should be set to at most a 1/3 of `sessionTimeout`'s value. Defaults to `"3s"`. | `"5s"` |

Reviewer 2: ```suggestion
| `heartbeatInterval` | N | Input | The interval between heartbeats to the consumer coordinator. At most, the value should be set to a 1/3 of the `sessionTimeout` value. Defaults to `"3s"`. | `"5s"` |
```

> | `sessionTimeout` | N | Input | The timeout used to detect client failures when using Kafka’s group management facility. If the broker fails to receive any heartbeats from the consumer before the expiration of this session timeout, then the consumer will be removed and initiate a rebalance. Defaults to `"10s"`. | `"20s"` |

Reviewer 2: ```suggestion
| `sessionTimeout` | N | Input | The timeout used to detect client failures when using Kafka’s group management facility. If the broker fails to receive any heartbeats from the consumer before the expiration of this session timeout, then the consumer is removed and initiates a rebalance. Defaults to `"10s"`. | `"20s"` |
```
